In [11]:
# estas son las librerías que vamos a usar para el análisis de datos y visualización de estos
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Cargamos el dataset de csv a un dataframe de pandas
df = pd.read_csv("train.csv") 

# Copia del dataset para usar este y dejar el original
df_limpio = df.copy()

#  aqui limpie los nulls lo que hace es que si hay valores nulls se rellenan con la mediana de cada columna 
df_limpio["bathrooms"] = df_limpio["bathrooms"].fillna(df_limpio["bathrooms"].median())
df_limpio["bedrooms"] = df_limpio["bedrooms"].fillna(df_limpio["bedrooms"].median())
df_limpio["beds"] = df_limpio["beds"].fillna(df_limpio["beds"].median())
df_limpio["review_scores_rating"] = df_limpio["review_scores_rating"].fillna(df_limpio["review_scores_rating"].median())

# Verificar si hay valores nulos
df_limpio.isnull().sum()

id                            0
log_price                     0
property_type                 0
room_type                     0
amenities                     0
accommodates                  0
bathrooms                     0
bed_type                      0
cancellation_policy           0
cleaning_fee                  0
city                          0
description                   0
first_review              15864
host_has_profile_pic        188
host_identity_verified      188
host_response_rate        18299
host_since                  188
instant_bookable              0
last_review               15827
latitude                      0
longitude                     0
name                          0
neighbourhood              6872
number_of_reviews             0
review_scores_rating          0
thumbnail_url              8216
zipcode                     966
bedrooms                      0
beds                          0
dtype: int64

Después del proceso de limpieza las variables importantes numericas ya no tiene nulos algunas si pero no son relevantes para el analisis o graficas que se realizaran

Algunas variables categóricas y de texto aún presentan valores faltantes, pero no afectan el análisis actual, ya que no son utilizadas en las visualizaciones principales.

In [12]:
#  esto muestra informacion general del dataframe

df_limpio.info() 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 74111 entries, 0 to 74110
Data columns (total 29 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   id                      74111 non-null  int64  
 1   log_price               74111 non-null  float64
 2   property_type           74111 non-null  object 
 3   room_type               74111 non-null  object 
 4   amenities               74111 non-null  object 
 5   accommodates            74111 non-null  int64  
 6   bathrooms               74111 non-null  float64
 7   bed_type                74111 non-null  object 
 8   cancellation_policy     74111 non-null  object 
 9   cleaning_fee            74111 non-null  bool   
 10  city                    74111 non-null  object 
 11  description             74111 non-null  object 
 12  first_review            58247 non-null  object 
 13  host_has_profile_pic    73923 non-null  object 
 14  host_identity_verified  73923 non-null

In [13]:
# Estadísticas descriptivas del dataframe
df_limpio.describe()

,id,log_price,accommodates,bathrooms,latitude,longitude,number_of_reviews,review_scores_rating,bedrooms,beds
count,7.411100e+04,74111.000000,74111.000000,74111.000000,74111.000000,74111.000000,74111.000000,74111.000000,74111.000000,74111.000000
mean,1.126662e+07,4.782069,3.155146,1.234628,38.445958,-92.397525,20.900568,94.503434,1.265467,1.709611
std,6.081735e+06,0.717394,2.153589,0.581386,3.080167,21.705322,37.828641,6.943159,0.851671,1.253389
min,3.440000e+02,0.000000,1.000000,0.000000,33.338905,-122.511500,0.000000,20.000000,0.000000,0.000000
25%,6.261964e+06,4.317488,2.000000,1.000000,34.127908,-118.342374,1.000000,93.000000,1.000000,1.000000
50%,1.225415e+07,4.709530,2.000000,1.000000,40.662138,-76.996965,6.000000,96.000000,1.000000,1.000000
75%,1.640226e+07,5.220356,4.000000,1.000000,40.746096,-73.954660,23.000000,99.000000,1.000000,2.000000
max,2.123090e+07,7.600402,16.000000,8.000000,42.390437,-70.985047,605.000000,100.000000,10.000000,18.000000



A partir de las estadísticas descriptivas se observan los siguientes puntos:


- El precio (log_price) tiene una media de 4.78 y una mediana de 4.70, lo que indica que la distribución es relativamente equilibrada, sin sesgos extremos.
- La capacidad promedio (accommodates) es de 3 personas, aunque existen propiedades con capacidad de hasta 16, lo que muestra alta variabilidad.
- La mayoría de las propiedades tienen 1 baño, 1 habitación y entre 1 y 2 camas, lo que indica predominancia de alojamientos pequeños.
- El número de reseñas presenta una gran dispersión (0 a 605), lo que sugiere diferencias significativas en popularidad entre propiedades.
- Las calificaciones son altas en general (media de 94.5), lo que indica buena satisfacción de los usuarios.

In [15]:
# la mediana para solo numericos
df_limpio.median(numeric_only=True)

id                      1.225415e+07
log_price               4.709530e+00
accommodates            2.000000e+00
bathrooms               1.000000e+00
cleaning_fee            1.000000e+00
latitude                4.066214e+01
longitude              -7.699696e+01
number_of_reviews       6.000000e+00
review_scores_rating    9.600000e+01
bedrooms                1.000000e+00
beds                    1.000000e+00
dtype: float64

La mediana muestra el valor central de cada variable:

- El precio (log_price) tiene una mediana de 4.70, lo que indica el valor típico del mercado.
- La capacidad (accommodates) es de 2 personas, mostrando que la mayoría de los alojamientos son pequeños.
- La mayoría de las propiedades tienen 1 baño, 1 habitación y 1 cama.
- El número de reseñas tiene una mediana de 6, lo que indica baja actividad en muchas propiedades.
- La calificación tiene una mediana de 96, reflejando alta satisfacción de los usuarios.

La mediana confirma que predominan propiedades pequeñas, bien calificadas y con poca cantidad de reseñas, lo cual puede influir en el precio.

Variables que influyen en el precio
las siguientes variables pueden influir en el precio

accommodates: indica la cantidad de personas que puede alojar la propiedad.
bedrooms: el numero de cuartos que tiene la propiedad
bathrooms: número de baños que tiene que ver con la comodidad y tamaño
beds: número de camas disponibles en los cuartos y propiedad
room_type: tipo de alojamiento  que son entero, privado o compartido.
city: la ubicacion de la propiedad puede influir en el precio ya que las casas en algunas ciudades pueden ser mas caras que en otras
Estas variables están relacionadas con el tamaño, capacidad y ubicación de la propiedad, por lo que es probable que tengan mucha influencia en el precio de los airbnbs.
